cliquet/tests/resource/test_views.py

```diff
@@ -56,31 +66,117 @@ def test_all_views_require_authentication(self):
+
+    def setUp(self):
+        super(RecordAuthzTest, self).setUp()
+        self.app.app.registry.permission.add_principal_to_ace(
+            self.collection_url, 'mushroom:create', self.principal)
 
         resp = self.app.post_json(self.collection_url,
-                                  MINIMALIST_RECORD)
-        self.assertEqual(permission_required(), 'readwrite')
+                                  MINIMALIST_RECORD,
+                                  headers=self.headers)
+        self.record = resp.json
+        self.record_url = self.get_item_url()
+        self.unknown_record_url = self.get_item_url(uuid.uuid4())
+
+    def test_record_get_is_denied_when_not_authorized(self):
+        self.app.get(self.record_url, headers=self.headers, status=403)
+
+    def test_record_patch_is_denied_when_not_authorized(self):
+        self.app.patch_json(self.record_url, MINIMALIST_RECORD,
+                            headers=self.headers, status=403)
+
+    def test_record_put_is_denied_when_not_authorized(self):
+        self.app.put_json(self.record_url, MINIMALIST_RECORD,
+                          headers=self.headers, status=403)
+
+    def test_record_delete_is_denied_when_not_authorized(self):
+        self.app.delete(self.record_url, headers=self.headers, status=403)
+
+    def test_record_get_is_granted_when_authorized(self):
+        object_id = self.record_url
+        self.app.app.registry.permission.add_principal_to_ace(
+            object_id, 'read', self.principal)
+
+        self.app.get(self.record_url, headers=self.headers, status=200)
+
+    def test_record_patch_is_granted_when_authorized(self):
+        object_id = self.record_url
+        self.app.app.registry.permission.add_principal_to_ace(
+            object_id, 'write', self.principal)
+
+        self.app.patch_json(self.record_url, MINIMALIST_RECORD,
+                            headers=self.headers, status=200)
+
+    def test_record_delete_is_granted_when_authorized(self):
+        object_id = self.record_url
+        self.app.app.registry.permission.add_principal_to_ace(
+            object_id, 'write', self.principal)
+
+        self.app.delete(self.record_url, headers=self.headers, status=200)
+
+    def test_record_put_on_existing_record_is_granted_when_authorized(self):
+        object_id = self.record_url
+        self.app.app.registry.permission.add_principal_to_ace(
+            object_id, 'write', self.principal)
+
+        self.app.put_json(self.record_url, MINIMALIST_RECORD,
+                          headers=self.headers, status=200)
 
-        url = self.item_url.format(id=resp.json['id'])
-        self.app.get(url)
-        self.assertEqual(permission_required(), 'readonly')
+    def test_record_put_on_unexisting_record_is_granted_when_authorized(self):
+        object_id = self.collection_url
+        self.app.app.registry.permission.add_principal_to_ace(
+                object_id, 'mushroom:create', self.principal)
```

Comment:
The permission name change each time and it is what makes the test understandable.